### DISCUSSION TITLE
What example should I use for a console app?

### DISCUSSION THREAD

**Discussion by jcollum-nutrien** [dotansimha/graphql-code-generator#8551] [Q&A]
I picked wrong, started with https://the-guild.dev/graphql/codegen/docs/guides/react-vue. But I really just want a simple standalone script that will use generated TS code to query a GQL endpoint and dump the data to the console. No React or web page needed, just a simple prototype. I don't see any examples that do that.

**Comment by charlypoly** [comment]
Hi @jcollum-nutrien,

Please refer to the documentation for ["Programmatic Usage"](https://www.the-guild.dev/graphql/codegen/docs/advanced/programmatic-usage) of codegen.

**Reply by jcollum-nutrien** [reply]
I think you're misunderstanding the question. I want to: 

1. generate a set of queries based on a remote endpoint (got this working) 
2. use those queries to pull data with a console app 

e.g.: 

1. generate a TS file for `getCustomerById` endpoint
2. use Apollo client (ideally) to run a query to `getCustomerById` with an input ID of like 5 or something
3. dump the result of the query to console with `console.table`:  
  ```
┌─────────┬─────────────────────┐
│ (index) │       Values        │
├─────────┼─────────────────────┤
│   id    │          5          │
│  name   │ 'Spacely Sprockets' │
│ planet  │      'Jupiter'      │
└─────────┴─────────────────────┘
```

Looking for a basic proof-of-concept console app that is easy to run and has minimal dependencies.

**Reply by charlypoly** [reply]
I would recommend you to use [`graphql-request`](https://github.com/prisma-labs/graphql-request) as GraphQL Client (lightweight) in combination with the `client-preset` setup (an example [is available here](https://github.com/dotansimha/graphql-code-generator/tree/master/examples/front-end/react/graphql-request) - using React but the setup is the same for a pure Node.js project)

**Reply by jcollum-nutrien** [reply]
Closer, but I'm getting a TS failure: 


```

import {request} from 'graphql-request'
import {graphql} from './generated/nutrien-dev-data'


const recommendationQuery = graphql(
  `
    query RecommendationById($input: RecommendationQueryInput!) {
        recommendationById(input: $input) {
            recId
            type
            branchId
            growerId
        }
    }
`
);

const endpoint = 'redacted';

const run = async () => {
  const headers = {};

  request({
    url: endpoint,

    document: recommendationQuery,  // <===== failure here 

    variables: {recId: '936b76c9-49cc-4cbb-80d1-88c7dc6ad7c3'} as RecommendationQueryInput,
    requestHeaders: headers,
  }).then((data) => console.table(data))
}
```

TS error: 

```
TS2322: Type 'unknown' is not assignable to type 'RequestDocument | TypedDocumentNode<any, 
RecommendationQueryInput>'.  types.d.ts(54, 5): The expected type comes from property 'document' which is declared 
here on type 'RequestExtendedOptions<RecommendationQueryInput, any>'
```

Sorry, I'm just not very good with TS. I don't see what the issue is.

**Reply by charlypoly** [reply]
Could you try to:
- make sure that codegen is running in watch mode (`with the -w` CLI flag)
- restart your VSCode TypeScript server (with the Cmd+Maj+P -> TypeScript: Restart server)

**Reply by jcollum-nutrien** [reply]
the schema is on a remote server, will -w poll the remote server for changes? 

I'm not using VSCode

I made a sample repo using a different GQL endpoint. The demo code works fine but the generated code has the same types twice. I submitted a bug for it. 

https://github.com/jcollum-nutrien/gql-code-generation-olympus

**Reply by charlypoly** [reply]
Hi @jcollum-nutrien,

The `client-preset` should not be combined with `typescript-*` plugins.
https://github.com/jcollum-nutrien/gql-code-generation-olympus/blob/0651bc764d8b4f293781a4e4ed70535eb0b4187b/config/faker-olympus-codegen.ts#L11-L12

**Reply by jcollum-nutrien** [reply]
I think that should be documented somewhere. Or is it and I missed it?

**Reply by charlypoly** [reply]
@jcollum-nutrien is not documented but since `@graphql-codegen/client-preset@1.1.0`, you will get errors when using `client-preset` with `typescript*` plugins.
